dpctl/tensor/_types.pxi

```diff
@@ -143,9 +144,11 @@ cdef int dtype_to_typenum(dtype) except *:
     else:
         try:
             dt = np.dtype(dtype)
-            if hasattr(dt, 'descr'):
-                return descr_to_typenum(dt)
-            else:
-                return -1
+        except TypeError:
```

Comment:
I believe `except *` in the definition of `dtype_to_typenum` isn't needed anymore also, since no exception is raised by the function.

Reply:
Correct. This is an optimization change though. It does not hurt to keep it.

I will push the suggested change and merge.